### DISCUSSION TITLE
working with DomPDF

### DISCUSSION THREAD

**Discussion by qcol** [awcodes/filament-tiptap-editor#123] [Q&A]
Hi
First of all, thank you for this great plugin. 
Is there any way to make the HTML code generated by the editor work with DomPDF? The styles don't work, even when I insert the content of plugin-tiptap-editor-2.5.0.0.css between <style>....</style> in the generated code.

**Comment by awcodes** [comment]
I honestly don't know. I've never used DomPDF.

**Comment by awcodes** [comment]
Should be releasing a version later today though that might help.

**Comment by qcol** [comment]
The update has not brought any changes. I'm not sure what the reason is. Presumably DomPDF needs full style information to generate the document as I see it on screen. 

 In most cases there will be a need to print the document created, working with any PDF generator (not necessarily DomPDF). Is there any other tested way to create printouts to look like they do on screen? (grid, tables, fonts, aligns etc.)

**Reply by awcodes** [reply]
In all honesty, after looking at dompdf's readme, it seems really behind the times and lacks a lot of css3 features which have been standard for a long time now.

Personally, I'd probably set up a route and view. Then use the tiptap_converter() helper to output the html in that blade. With that set up you could use Spatie's browser shot package to generate the pdf. Might give better output since it's using puppeteer to load an actual browser rendered page.

**Reply by qcol** [reply]
Thanks a lot, I will try

**Reply by qcol** [reply]
I've tried generating a simple blade file as per your suggestion but have no idea why - the styles are not applied to either the grid or the table. What am I doing wrong? 

![image](https://github.com/awcodes/filament-tiptap-editor/assets/358993/4a1cc152-fbbc-4c91-a2d3-ea9f5da5179c)

**Reply by awcodes** [reply]
Did you even look at that stylesheet? All those styles are scoped to the editor. You need to create your own stylesheet for front end display.

**Reply by qcol** [reply]
I looked but compressed so poor readability :) I thought that outside the editor they also format used elements correctly. 
OK, so the issue has been resolved. Thank you for your help.
